src/Core/Checkout/Order/Aggregate/OrderLineItem/OrderLineItemCollection.php

```diff
@@ -76,8 +78,27 @@ public function filterGoodsFlat(): array
         return $filtered;
     }
 
+    public function hasLineItemWithType(string $type): bool
+    {
+        foreach ($this->buildFlat($this) as $lineItem) {
+            if ($lineItem->hasPayloadValue(LineItem::PAYLOAD_PRODUCT_TYPE) && $type === $lineItem->getPayloadValue(LineItem::PAYLOAD_PRODUCT_TYPE)) {
```

Comment:
`getPayloadValue` is also calling `hasPayloadValue`. so we can probably just directly call `getPayloadValue` safely. also check other places